src/components/connect-request.js

```diff
@@ -50,39 +67,77 @@ const connectRequest = (mapPropsToConfig, options = {}) => (WrappedComponent) =>
             return this._wrappedInstance;
         }
 
-        cancelPendingRequests() {
+        // returns [[ "keys", "to", "cancel"], [{ config objects to request }]]
+        _diffConfigs(prevConfigs, configs) {
+            const unpack = (c) => {
+                return getQueryKey(c.url, c.body);
+            };
+            const prevQueryKeys = prevConfigs.map(unpack);
+            const queryKeys = configs.map(unpack);
+            const i = intersection(prevQueryKeys, queryKeys);
+            const cancelKeys = difference(prevQueryKeys, i);
+            const requestKeys = difference(queryKeys, i);
+
+            const toRequest = prevConfigs.filter((c) => {
+                const k = getQueryKey(c.url, c.body);
+                return requestKeys.includes(k);
+            });
+
+            return [cancelKeys, toRequest];
```

Comment:
Instead of using an array for tuple-like behavior, can you change to use a regular object instead and destructure from the call site in `componentDidUpdate`?